### PR TITLE
ios simulator arm64 and arm64e support

### PR DESCRIPTION
**Pull request by swasti29**
Update cmake to consider platform passed as argument for building 1DS for ios simulator m1(arm64 and arm64e) architecture

### CODE REVIEW

File: build-ios.sh
Comment (oldalton) [inline build-ios.sh:33]: Why was this changed?

File: CMakeLists.txt
Comment (lalitb) [inline CMakeLists.txt:62]: Instead of adding lines from 53 to 62, can we just add ${IOS_PLAT} at lines 67 and 72;
```cmake
    elseif(${IOS_ARCH} STREQUAL "arm64")
          set(IOS_PLATFORM ${IOS_PLAT})
           ...
    elseif(${IOS_ARCH} STREQUAL "arm64e")
           set(IOS_PLATFORM ${IOS_PLAT})
            ...
    else()
```

File: CMakeLists.txt
Comment (lalitb) [inline CMakeLists.txt:256]: we can keep this as default "YES" ?

File: build-ios.sh
Comment (lalitb) [inline build-ios.sh:26]: When is "$4" be used, as there seems to be only 3 arguments - <release_type>, <arch_type>, <device_type>

Comment (swasti29) [reply build-ios.sh:26]: There are 4 arguments 
**clean** <release_type>, <arch_type>, <device_type>

File: build-ios.sh
Comment (swasti29) [reply build-ios.sh]: Will add another PR to pass ios deployment target as argument.

**Review by lalitb — APPROVED** [review]
Changes look good to me. But it is important to get the approval from @eduardo-camacho before merging it. As mentioned in the mail discussion:

---
Please make sure to have tested following scenarios (actually linking library and makes sure it works; lipo checking might not be enough):
1.	Simulator on M1
2.	Simulator on Intel
3.	Physical device on M1
4.	Physical device on Intel
---

**Review by sid-dahiya — APPROVED** [review]

**Comment by eduardo-camacho** [comment]
> Changes look good to me. But it is important to get the approval from @eduardo-camacho before merging it. As mentioned in the mail discussion:
> 
> Please make sure to have tested following scenarios (actually linking library and makes sure it works; lipo checking might not be enough):
> 
> 1. Simulator on M1
> 2. Simulator on Intel
> 3. Physical device on M1
> 4. Physical device on Intel

Do you have an update on testing scenarios above?

File: CMakeLists.txt
Comment (eduardo-camacho) [inline CMakeLists.txt:39]: This is the same line than 35, not needed

**Review by eduardo-camacho — APPROVED** [review]
Left a nit

File: build.sh
Comment (lalitb) [inline build.sh:95]: This part of code can be simplified by letting user optionally set the env-variable `MACOSX_DEPLOYMENT_TARGET` (just like CMAKE_OPTS and GIT_PULL_TOKEN) from shell/terminal before invoking this script. And then we can just write:

```bash
default_mac_os_target=$([ "$MAC_ARCH" == "arm64" ] && echo "11.10" || echo "10.10")
[ -z "$MACOSX_DEPLOYMENT_TARGET" ] && export MACOSX_DEPLOYMENT_TARGET="${default_mac_os_target}"
echo "macosx deployment target="$MACOSX_DEPLOYMENT_TARGET
```

The help comment above can be changed to mention `MACOSX_DEPLOYMENT_TARGET` under env-variable section

File: build-ios.sh
Comment (lalitb) [inline build-ios.sh:37]: nit - fix the indentation

File: CMakeLists.txt
Comment (lalitb) [inline CMakeLists.txt:28]: The cmake command can be invoked directly from shell/terminal, and in that case default value for FORCE_RESET_OSX_DEPLOYMENT_TARGET should remain unchanged. Probably change above code to:

```cmake
option(FORCE_RESET_OSX_DEPLOYMENT_TARGET "Clear the OSX Deployment Target Set" YES)
if (DEFINED FORCE_RESET_DEPLOYMENT_TARGET)
  set(FORCE_RESET_OSX_DEPLOYMENT_TARGET, $FORCE_RESET_DEPLOYMENT_TARGET)
endif()
```

**Review by lalitb — COMMENTED** [review]
Have few comments.

**Comment by swasti29** [comment]
> Changes look good to me. But it is important to get the approval from @eduardo-camacho before merging it. As mentioned in the mail discussion:
> 
> Please make sure to have tested following scenarios (actually linking library and makes sure it works; lipo checking might not be enough):
> 
> 1. Simulator on M1
> 2. Simulator on Intel
> 3. Physical device on M1
> 4. Physical device on Intel

I have tested with and without optional deployment target paramter by linking library on Simulator and Physical device with both M1 and Intel

**Comment by swasti29** [comment]
> I have tested with and without optional deployment target paramter by linking library on Simulator and Physical device with both M1 and Intel



> Have few comments.

Resolved

**Comment by swasti29** [comment]
> > Changes look good to me. But it is important to get the approval from @eduardo-camacho before merging it. As mentioned in the mail discussion:
> > Please make sure to have tested following scenarios (actually linking library and makes sure it works; lipo checking might not be enough):
> > 
> > 1. Simulator on M1
> > 2. Simulator on Intel
> > 3. Physical device on M1
> > 4. Physical device on Intel
> 
> Do you have an update on testing scenarios above?

I have tested with and without optional deployment target paramter by linking library on Simulator and Physical device with both M1 and Intel.

**Comment by lalitb** [comment]
@swasti29  - Changes look good now. You should be able to merge this PR once CI checks are completed ( you may have rerun the check incase of transient failures ).

**Comment by eduardo-camacho** [comment]
> @swasti29 - Changes look good now. You should be able to merge this PR once CI checks are completed ( you may have to rerun the CI check incase of transient failures ).

I approved yesterday night, we are good to go. Thanks!

**Pull request merged**
